src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -23,9 +23,9 @@
          hostcrl           (get-in config [:puppet-server :hostcrl])
          settings          (ca/config->master-settings config)
          jruby-service     (tk-services/get-service this :JRubyPuppetService)
-         upgrade-error     (core/construct-404-error-message jruby-service)
-         product-name      (get-in config [:product :name])
-         update-server-url (get-in config [:product :update-server-url])]
+         product-name      (or (get-in config [:product :name]) {:group-id "puppetlabs" :artifact-id "puppet-server"})
+         upgrade-error     (core/construct-404-error-message jruby-service product-name)
+         update-server-url (or (get-in config [:product :update-server-url]) "http://updates.puppetlabs.com")]
```

Comment:
Do we want/need to code the "http://updates.puppetlabs.com" default in here?  If it is passed through as nil, doesn't the dujour-version-check library use "http://updates.puppetlabs.com" by default?  Wasn't sure we wanted to have two levels of defaults - in puppet-server and the underlying library - for the case that it doesn't appear in the config file.